diff --git a/daemon/gdm-launch-environment.c b/daemon/gdm-launch-environment.c
index 110520c0b..5739e8c3d 100644
--- a/daemon/gdm-launch-environment.c
+++ b/daemon/gdm-launch-environment.c
@@ -277,6 +277,7 @@ build_launch_environment (GdmLaunchEnvironment *launch_environment,
                 "WINDOWPATH",
                 "XCURSOR_PATH",
                 "XDG_CONFIG_DIRS",
+                "GUIX_GDK_PIXBUF_MODULE_FILES",
                 NULL
         };
         g_autofree char *system_data_dirs = NULL;
