Unbundles sqlitecpp dependency.

patch by bdunahu <bdunahu@operationnull.com>

diff --git a/3rdparty/CMakeLists.txt b/3rdparty/CMakeLists.txt
index 46c0c26e1..d4ea7d860 100644
--- a/3rdparty/CMakeLists.txt
+++ b/3rdparty/CMakeLists.txt
@@ -68,18 +68,7 @@ set(SQLITE_HAS_CODEC OFF CACHE BOOL "Enable database encryption API. Not availab
 if(SQLite3_VERSION VERSION_LESS 3.19)
 	set(SQLITE_USE_LEGACY_STRUCT ON CACHE BOOL "Fallback to forward declaration of legacy struct sqlite3_value (pre SQLite 3.19)" FORCE)
 endif()
-set(SQLITECPP_INTERNAL_SQLITE OFF CACHE BOOL "Add the internal SQLite3 source to the project." FORCE)
-set(SQLITECPP_INCLUDE_SCRIPT OFF CACHE BOOL "Include config & script files." FORCE)
-set(SQLITECPP_RUN_CPPLINT OFF CACHE BOOL "Run cpplint.py tool for Google C++ StyleGuide." FORCE)
-set(SQLITECPP_RUN_CPPCHECK OFF CACHE BOOL "Run cppcheck C++ static analysis tool." FORCE)
-set(SQLITECPP_BUILD_EXAMPLES OFF CACHE BOOL "Build examples." FORCE)
-set(SQLITECPP_BUILD_TESTS OFF CACHE BOOL "Build and run tests." FORCE)
-set(SQLITECPP_INSTALL OFF CACHE BOOL "Enables the install target." FORCE)
-if(CMAKE_SYSTEM_NAME MATCHES "Emscripten")
-	set(SQLITECPP_USE_STACK_PROTECTION OFF CACHE BOOL "USE Stack Protection hardening." FORCE)
-endif()
-add_subdirectory(SQLiteCpp EXCLUDE_FROM_ALL)
-set_property(TARGET SQLiteCpp PROPERTY FOLDER "3rdparty")
+find_package(SQLiteCpp REQUIRED)
 
 # glad library
 
